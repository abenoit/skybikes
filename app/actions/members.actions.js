import { SIGN_UP, LOGIN_SUCCESS, LOGOUT } from "../constants/actionTypes";

export const login = email => {
  return { type: LOGIN_SUCCESS, email };
};

export const logout = () => {
  return { type: LOGOUT };
};

export const signup = (firstname, lastname, email, phone) => {
  return { type: SIGN_UP, firstname, lastname, email, phone };
};
