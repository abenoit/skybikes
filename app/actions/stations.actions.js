import { RENT_BIKE, RETURN_BIKE } from "../constants/actionTypes";

export const rentBike = stationId => {
  return {
    type: RENT_BIKE,
    stationId
  };
};

export const returnBike = stationId => {
  return {
    type: RETURN_BIKE,
    stationId
  };
};
