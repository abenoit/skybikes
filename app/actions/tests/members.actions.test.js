import { login, logout, signup } from "../members.actions";

describe("Members Actions", () => {
  it("login", () => {
    const member = {
      email: "test@test.com"
    };

    expect(login(member)).toMatchSnapshot();
  });

  it("logout", () => {
    expect(logout()).toMatchSnapshot();
  });

  it("signup", () => {
    expect(
      signup("Rainbow", "Dash", "rainbow@dash.com", "0102030405")
    ).toMatchSnapshot();
  });
});
