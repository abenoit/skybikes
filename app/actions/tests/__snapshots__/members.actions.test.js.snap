// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Members Actions login 1`] = `
Object {
  "email": Object {
    "email": "test@test.com",
  },
  "type": "LOGIN_SUCCESS",
}
`;

exports[`Members Actions logout 1`] = `
Object {
  "type": "LOGOUT",
}
`;

exports[`Members Actions signup 1`] = `
Object {
  "email": "rainbow@dash.com",
  "firstname": "Rainbow",
  "lastname": "Dash",
  "phone": "0102030405",
  "type": "SIGN_UP",
}
`;
